modules/tensor_mechanics/src/postprocessors/InteractionIntegral.C

```diff
@@ -0,0 +1,346 @@
+                                "the plane of the crack, normal to the specified "
+                                "axis (0=x, 1=y, 2=z)");
+  params.addParam<bool>("t_stress", false, "Calculate T-stress");
+  params.addParam<Real>("poissons_ratio", "Poisson's ratio for the material.");
+  params.addParam<Real>("youngs_modulus", "Young's modulus of the material.");
+  params.set<bool>("use_displaced_mesh") = false;
+  params.addParam<unsigned int>("ring_index", "Ring ID");
+  params.addParam<unsigned int>("ring_first", "First Ring ID");
+  MooseEnum q_function_type("Geometry Topology", "Geometry");
+  params.addParam<MooseEnum>("q_function_type",
+                             q_function_type,
+                             "The method used to define the integration domain. Options are: " +
+                                 q_function_type.getRawNames());
+  MooseEnum sif_modes("KI KII KIII T", "KI");
+  params.addRequiredParam<MooseEnum>("sif_mode",
+                                     sif_modes,
+                                     "Stress intensity factor to Calculate. Choices are: " +
+                                         sif_modes.getRawNames());
+
+  return params;
+}
+
+InteractionIntegral::InteractionIntegral(const InputParameters & parameters)
+  : ElementIntegralPostprocessor(parameters),
+    _crack_front_definition(&getUserObject<CrackFrontDefinition>("crack_front_definition")),
+    _has_crack_front_point_index(isParamValid("crack_front_point_index")),
+    _crack_front_point_index(
+        _has_crack_front_point_index ? getParam<unsigned int>("crack_front_point_index") : 0),
+    _treat_as_2d(false),
+    _stress(getMaterialPropertyByName<RankTwoTensor>("stress")),
+    _strain(getMaterialPropertyByName<RankTwoTensor>("elastic_strain")),
+    _grad_disp_x(coupledGradient("disp_x")),
+    _grad_disp_y(coupledGradient("disp_y")),
+    _grad_disp_z(parameters.get<SubProblem *>("_subproblem")->mesh().dimension() == 3
+                     ? coupledGradient("disp_z")
+                     : _grad_zero),
+    _has_temp(isCoupled("temp")),
+    _grad_temp(_has_temp ? coupledGradient("temp") : _grad_zero),
+    _current_instantaneous_thermal_expansion_coef(
+        hasMaterialProperty<Real>("current_instantaneous_thermal_expansion_coef")
+            ? &getMaterialProperty<Real>("current_instantaneous_thermal_expansion_coef")
+            : NULL),
+    _K_factor(getParam<Real>("K_factor")),
+    _has_symmetry_plane(isParamValid("symmetry_plane")),
+    _t_stress(getParam<bool>("t_stress")),
+    _poissons_ratio(getParam<Real>("poissons_ratio")),
+    _youngs_modulus(getParam<Real>("youngs_modulus")),
+    _ring_index(getParam<unsigned int>("ring_index")),
+    _q_function_type(getParam<MooseEnum>("q_function_type")),
+    _sif_mode(getParam<MooseEnum>("sif_mode"))
+{
+  if (_has_temp && !_current_instantaneous_thermal_expansion_coef)
+    mooseError("To include thermal strain term in interaction integral, must both couple "
```

Comment:
@bwspenc, does this error need to be present in the TM version as well? I am not able to run the InteractionIntegral tests inside domain_integral_thermal due to this error and I am not sure what needs to be added to the material models to fix this error.

Reply:
Instead, we'll need to get this using `getMaterialPropertyDerivative` for the derivative of all the eigenstrains with respect to temperature (as is done in the EshelbyTensor material).